jctools-core/src/main/java/org/jctools/queues/MpmcArrayQueue.java

```diff
@@ -144,7 +148,8 @@ public boolean offer(final E e) {
             pIndex = lvProducerIndex();
             seqOffset = calcSequenceOffset(pIndex, mask);
             seq = lvSequence(sBuffer, seqOffset);
-            if (seq < pIndex) { // consumer has not moved this seq forward, it's as last producer left
```

Comment:
Moved this comment because of some bugs in javaparser meant this was a work around to preserve it